app/controllers/buses_controller.rb

```diff
@@ -0,0 +1,53 @@
+require 'net/http'
+require 'json'
+
+class BusesController < ApplicationController
+  def index
+  end
+
+  def stop
+    stop = params[:stop]
+    @arrivals = fetch_arrivals([stop])
+    render :arrivals
+  end
+
+  def postcode
+    postcode = params[:postcode]
+    lat, lon = fetch_postcode_coordinates(postcode)
+    stops = fetch_nearest_stops(lat, lon)
+    @arrivals = fetch_arrivals(stops)
+    render :arrivals
+  end
+
+  private
+
+  def fetch_arrivals(stops)
+    arrivals = []
+
+    stops.each do |stop|
+      uri = URI("https://api.tfl.gov.uk/StopPoint/#{stop}/Arrivals")
```

Comment:
I think it's better to build the URI, e.g. URI::HTTP.build(host: 'https://api.tfl.gov.uk', path: '/StopPoint/#{stop}/Arrivals')
https://ruby-doc.org/stdlib-2.5.1/libdoc/uri/rdoc/URI/HTTP.html

That way you can also pull out your base url into a constant - if TFL ever changed that url and you use it for multiple calls, you'd only need to change it in one place